mundane-bytes/src/main/scala/com/ambiata/mundane/bytes/Buffer.scala

```diff
@@ -0,0 +1,26 @@
+package com.ambiata.mundane.bytes
+
+case class Buffer(bytes: Array[Byte], offset: Int, var length: Int) {
+
+  /** Ensure there is enough space in the array, and if not return a _new_ [[Buffer]] and array with the original values */
+  def allocate(i: Int): Buffer =
+    if (length + i <= bytes.length) this
+    else {
+      val b = new Buffer(new Array(math.max(bytes.length + i, bytes.length * 2)), offset, length)
+      // TODO Copy the entire array, or just from the offset?
```

Comment:
@markhibberd I'm not sure about this - it seems like consumers have more options if we ignore the offset here, but am I missing some danger/correctness issue?